src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,69 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String PARAMETER_0_SUPPLY = "supply";
+    private static final String PARAMETER_1_BUY = "buy";
+    private static final String PARAMETER_2_RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] incomeData = getIncomeData(fromFileName);
+        int[] reportNumbers = getReportNumbers(incomeData);
+        writeReportToFile(toFileName, reportNumbers);
+    }
+
+    private String[] getIncomeData(String fromFileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+        String[] resultData;
+        File incomeData = new File(fromFileName);
+        try (BufferedReader reader = new BufferedReader(new FileReader(incomeData))) {
+            String lineValue = reader.readLine();
+            while (lineValue != null) {
+                stringBuilder.append(lineValue).append(",");
+                System.out.println(lineValue);
+                lineValue = reader.readLine();
+            }
+            resultData = stringBuilder.toString().split(",");
+        } catch (IOException e) {
+            throw new RuntimeException("Can't open file", e);
+        }
+        return resultData;
+    }
+
+    private int[] getReportNumbers(String[] incomeData) {
+        int supply = 0;
+        int buy = 0;
+        int result;
+        for (int i = 0; i < incomeData.length; i = i + 2) {
+            if (incomeData[i].equals(PARAMETER_0_SUPPLY)) {
+                supply = supply + Integer.parseInt(incomeData [i + 1]);
+                continue;
+            }
+            if (incomeData[i].equals(PARAMETER_1_BUY)) {
+                buy = buy + Integer.parseInt(incomeData[i + 1]);
+            }
+        }
+        result = supply - buy;
+        return new int[] {supply, buy, result};
+    }
```

Comment:
While returning an array is a valid approach, it could potentially introduce the need for magic numbers in the next method.

Alternatively, a more concise and readable solution would be to generate the report string right within this method, instead of using an array. This would eliminate the need for indexing and improve code readability.